### PR TITLE
feat(storage): support autoclass

### PR DESCRIPTION
**Pull request by diptanshumittal**

### CODE REVIEW

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb [hunk 1]
```diff
@@ -407,6 +416,40 @@ def storage_class= new_storage_class
...
+        #
+        def autoclass_toggle_time
+          @gapi.autoclass&.toggle_time
+        end
+
+        ##
+        # Updates bucket's autoclass configuration. This defines the default class for objects in the
+        # bucket and down/up-grades the storage class of objects based on the access patterns.
+        # Accepted values are `:false`, and `:true`.
+        # rubocop:todo update the link
```
Comment (SandeepTuniki) [inline google-cloud-storage/lib/google/cloud/storage/bucket.rb:441]: What does this line do?

Comment (diptanshumittal) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb:441]: Todo reference for future.

Comment (cojenco) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb:441]: Just confirmed the docs link. Feel free to add it in. Thanks for your patience!

https://cloud.google.com/storage/docs/using-autoclass

**Comment by diptanshumittal** [comment]
Closing this pr. Refer to - #19210.

**Pull request closed without merging**